registry/registryprovider.go

```diff
@@ -166,44 +186,90 @@ func (grp githubRegistryProvider) createGithubClient(credentialName string) (*gi
 				&oauth2.Token{AccessToken: string(c.APIToken)},
 			)
 			tc := oauth2.NewClient(oauth2.NoContext, ts)
-			return github.NewClient(tc), nil
+			return tc, github.NewClient(tc), nil
 		}
 		if c.BasicAuth.Username != "" && c.BasicAuth.Password != "" {
 			tp := github.BasicAuthTransport{
 				Username: c.BasicAuth.Username,
 				Password: c.BasicAuth.Password,
 			}
-			return github.NewClient(tp.Client()), nil
+			return tp.Client(), github.NewClient(tp.Client()), nil
 		}
 
 	}
-	return nil, fmt.Errorf("No suitable credential found for %s", credentialName)
+	return nil, nil, fmt.Errorf("No suitable credential found for %s", credentialName)
 
 }
 
 // GetGithubRegistry returns a new GithubRegistry. If there's a credential that is specified, will try
 // to fetch it and use it, and if there's no credential found, will fall back to unauthenticated client.
 func (grp githubRegistryProvider) GetGithubRegistry(cr common.Registry) (GithubRegistry, error) {
-	if cr.Type == common.GithubRegistryType {
-		// If there's a credential that we need to use, fetch it and create a client for it.
-		client, err := grp.createGithubClient(cr.CredentialName)
-		if err != nil {
-			return nil, err
-		}
+	// If there's a credential that we need to use, fetch it and create a client for it.
+	httpClient, client, err := grp.createGithubClient(cr.CredentialName)
```

Comment:
If I'm not mistaken here (see previous comment for caveat), httpClient is derivable from client.